### PR TITLE
#349: Ask to optin and ack that users have permissions

### LINKED ISSUE #349: GDPR ask if permission to use people's email
Add a checkbox "I confirm that I have permission to use all the participants email addresses for Secret Santa". Should be implemented before may 2018

### PR DESCRIPTION
**Pull request by tvlooy**
closes #349

### CODE REVIEW

File: src/Intracto/SecretSantaBundle/Resources/meta/LICENSE [hunk 1]
```diff
@@ -1,22 +0,0 @@
...
-with or without fee is hereby granted, provided that the above copyright notice
-and this permission notice appear in all copies.
-
-THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
-REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND
-FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
-INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
-LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
-OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
-PERFORMANCE OF THIS SOFTWARE.
```
Comment (tvlooy) [inline src/Intracto/SecretSantaBundle/Resources/meta/LICENSE]: license zit in de root van het project dus remove hier

File: src/Intracto/SecretSantaBundle/Resources/views/Party/create.html.twig [hunk 2]
```diff
@@ -103,11 +103,18 @@
                         </button>
                     </div>
                 </div>
-                <h1>{{ 'party-create.add_personal_message.title'|trans }}</h1>
+                <h2>{{ 'party-create.add_personal_message.title'|trans }}</h2>
                 {{ form_widget(form.message, {'attr': {'rows': '8', 'class': 'field span12'} }) }}<br/>
```
Comment (tvlooy) [inline src/Intracto/SecretSantaBundle/Resources/views/Party/create.html.twig]: een page mag maar 1 h1 tag hebben dacht ik, dus even aangepast

File: src/Intracto/SecretSantaBundle/Resources/views/Party/create.html.twig [hunk 3]
```diff
@@ -142,12 +149,12 @@
...
-                    '<th class="col-xs-3 col-md-4">{{ "party-create.label.participants"|trans }}</th>' +
-                    '<th class="col-xs-3 col-md-4">{{ "party-create.label.email"|trans }}</th>' +
-                    '<th style="text-align: right;" class="col-xs-4">  </th> ' +
-                    '</tr>'
+                '<tr>' +
+                '<th class="col-xs-1 participant-number-header">#</th>' +
+                '<th class="col-xs-3 col-md-4">{{ "party-create.label.participants"|trans }}</th>' +
+                '<th class="col-xs-3 col-md-4">{{ "party-create.label.email"|trans }}</th>' +
+                '<th style="text-align: right;" class="col-xs-4">  </th> ' +
+                '</tr>'
```
Comment (tvlooy) [inline src/Intracto/SecretSantaBundle/Resources/views/Party/create.html.twig]: style

File: src/Intracto/SecretSantaBundle/Resources/views/Party/createCopy.html.twig [hunk 4]
```diff
@@ -31,10 +31,12 @@
...
-                Secret Santa, Kris Kringle, Kris Kindle, Kris Kringel, Wichteln, or even lootjes trekken, namen
-                trekken or Monito Monita: all over the world people use different names for the same thing. So
-                let us take you on a quick trip around the world! No worries, you’ll be home in time for your
-                <b>Secret Santa party</b>.
+                Secret Santa, Kris Kringle, Kris Kindle, Kris Kringel, White Elephant, Yankee Swap, Yankee Doodle,
+                Wichteln, or even lootjes trekken, namen trekken or Monito Monita: all over the world people use
+                different names for the same thing. So let us take you on a quick trip around the world! No worries,
+                you’ll be home in time for your <b>Secret Santa party</b>.
+
+                 is a gift exchange game that is very similar in style to the  gift exchange. Yankee Swap of yankee doodle
```
Comment (tvlooy) [inline src/Intracto/SecretSantaBundle/Resources/views/Party/createCopy.html.twig]: extra SEO woordjes

File: src/Intracto/SecretSantaBundle/Resources/views/Party/createCopy.html.twig [hunk 5]
```diff
@@ -31,10 +31,10 @@
...
-                Secret Santa, Kris Kringle, Kris Kindle, Kris Kringel, Wichteln, or even lootjes trekken, namen
-                trekken or Monito Monita: all over the world people use different names for the same thing. So
-                let us take you on a quick trip around the world! No worries, you’ll be home in time for your
-                <b>Secret Santa party</b>.
+                Secret Santa, Kris Kringle, Kris Kindle, Kris Kringel, White Elephant, Yankee Swap, Yankee Doodle,
+                Wichteln, or even lootjes trekken, namen trekken or Monito Monita: all over the world people use
+                different names for the same thing. So let us take you on a quick trip around the world! No worries,
+                you’ll be home in time for your <b>Secret Santa party</b>.
             </p>
             <ul>
```
Comment (tvlooy) [inline src/Intracto/SecretSantaBundle/Resources/views/Party/createCopy.html.twig]: extra SEO woordjes

**Comment by tvlooy** [comment]
@acrobat lijkt me niet echt elegant zo. Wat denk jij?

**Comment by acrobat** [comment]
@tvlooy ik zal straks eens door de changes gaan!

File: app/Resources/translations/messages.en.yml [hunk 6]
```diff
@@ -16,6 +16,7 @@ form-party:
         location: Location of your Secret Santa party
         amount_to_spend: Amount to spend
         message: Additional message in email
+        confirmed: Yes, I want to opt-in and have permission to use the participant's e-email addresses for Secret Santa
```
Comment (acrobat) [inline app/Resources/translations/messages.en.yml]: Typo 'e-email'

File: src/Intracto/Behat/Features/create_party.feature [hunk 7]
```diff
@@ -6,7 +6,8 @@ Feature: Create a new party
     When I create a party with 5 participants
     And I choose a party date in the future
     And I choose a location
-    And I choose the amount to spend
+	And I choose the amount to spend
```
Comment (acrobat) [inline src/Intracto/Behat/Features/create_party.feature]: Wrong indent

**Comment by acrobat** [comment]
Some small remarks. Two questions:

- Would it be better to save the DateTime of opt-in? This way we know if and when the owner did this opt-in
- Is it correct behavior to just opt them in on the update page? Or should we ask the explicitly?

**Comment by tvlooy** [comment]
if they arrive on the update page they are already opted in. From now on nobody can create when they don't opt-in so the opt in date is the date when the party was created ($creationdate)

**Pull request merged**